### PR TITLE
Fix: DIF issues

### PR DESCRIPTION
**Pull request by slesaad**
**Changes**
- Prioritized field dependencies to check dependencies
- Added collection `version` to collection datetime validation with granules for accuracy
- Allowed DIF10 datetime fields to support ISO Date (not just ISO Datetime)
- Generalized and renamed `datetime_compare` check to `date_compare`
- Updated auto GCMD keywords downloader to use the new GCMD url
- Added sensor number check

### CODE REVIEW

File: pyQuARC/code/datetime_validator.py [hunk 1]
```diff
@@ -58,6 +76,24 @@ def iso_format_check(datetime_string):
             "value": datetime_string,
         }
 
+    @staticmethod
+    @if_arg
+    def date_datetime_format_check(datetime_string):
```
Comment (xhagrg) [inline pyQuARC/code/datetime_validator.py:81]: `date_or_datetime_format_check`?

File: pyQuARC/code/scheduler.py [hunk 2]
```diff
@@ -27,13 +27,13 @@ def get_all_dependencies(self, rule_id, check, field_dict=None):
...
 
         if field_dict:
-            dependencies_from_fields = field_dict.get("dependencies", [])
-        else:
-            rule = self.rule_mapping.get(rule_id)
-            if field_objects := rule.get("fields_to_apply").get(self.metadata_format):
-                for field_object in field_objects:
-                    if field_dependencies := field_object.get("dependencies"):
-                        dependencies_from_fields.extend(field_dependencies)
+            return dependencies_from_fields
```
Comment (xhagrg) [inline pyQuARC/code/scheduler.py:30]: returning empty dependencies?

**Review by xhagrg — COMMENTED** [review]
Some small comments.

File: pyQuARC/code/scheduler.py (hunk 2 above)
Comment (slesaad) [reply pyQuARC/code/scheduler.py]: oopsie. https://github.com/NASA-IMPACT/pyQuARC/pull/190/commits/729ee36b79ea9ad00ec3bc1d18c07e575a760235

File: pyQuARC/code/datetime_validator.py [hunk 3]
```diff
@@ -66,8 +102,8 @@ def compare(first, second, relation):
         Returns:
             (dict) An object with the validity of the check and the instance
         """
-        first = DatetimeValidator._iso_datetime(first)
-        second = DatetimeValidator._iso_datetime(second)
+        first = DatetimeValidator._iso_datetime(first) or DatetimeValidator._iso_date(first)
```
[the remark this answers is not in the document]
Comment (slesaad) [reply pyQuARC/code/datetime_validator.py:105]: yeah,
dif supports both iso date and datetime while the others support only iso datetime
and there's no way to AND/OR the checks

File: pyQuARC/code/datetime_validator.py [hunk 4]
```diff
@@ -40,6 +40,24 @@ def _iso_datetime(datetime_string):
...
+        Args:
+            date_string (str): the datestring
+
+        Returns:
+            (datetime.datetime) If the string is valid iso string, False otherwise
+        """
+        
+        try:
+            value = datetime.strptime(date_string, "%Y-%m-%d")
+            return value
```
[the remark this answers is not in the document]
Comment (slesaad) [reply pyQuARC/code/datetime_validator.py:57]: we could do that, but we'll also need to make sure that the date is not invalid; like 2021-02-29 (2021 is not a leap year) or 2021-04-31 (April has only 30 days), etc, which this just does

File: pyQuARC/code/scheduler.py [hunk 5]
```diff
@@ -27,13 +27,13 @@ def get_all_dependencies(self, rule_id, check, field_dict=None):
...
-                for field_object in field_objects:
-                    if field_dependencies := field_object.get("dependencies"):
-                        dependencies_from_fields.extend(field_dependencies)
+            return field_dict.get("dependencies", [])
+
+        rule = self.rule_mapping.get(rule_id)
+        if field_objects := rule.get("fields_to_apply").get(self.metadata_format):
+            for field_object in field_objects:
+                if field_dependencies := field_object.get("dependencies"):
+                    dependencies_from_fields.extend(field_dependencies)
```
Comment (xhagrg) [inline pyQuARC/code/scheduler.py:36]: Do we not need to go through these steps even when the field_dict is present?

**Comment by xhagrg** [comment]
Left a minor comment, other than that it looks good.

File: pyQuARC/code/scheduler.py (hunk 5 above)
Comment (slesaad) [reply pyQuARC/code/scheduler.py:36]: no, field_dict contains the specific dependencies that relate to the given field, so we need only that
`field_dict` isn't always available
so we have to search through all the dependencies when it's not there, used when we need to order the calls to the rules based on the dependencies

**Comment by xhagrg** [comment]
Small comment, otherwise LGTM.

**Pull request merged**